src/main.rs

```diff
@@ -152,26 +138,30 @@ async fn main() -> Result<()> {
             notify_shutdown.clone(),
         ));
         loop {
-            config_reload.notified().await;
-            match Settings::from_file(&temp_path) {
-                Ok(new_settings) => {
-                    settings = new_settings;
-                    notify_shutdown.notify_waiters();
-                    notify_shutdown.notified().await;
-                    fs::rename(&temp_path, &config_path).unwrap_or_else(|e| {
-                        error!("Failed to rename the new configuration file: {e}");
-                    });
-                    break;
+            if let Some(config) = config_rx.recv().await {
+                if args.config.is_some() {
+                    warn!("Cannot update the configuration from the Manager server when a local configuration file is specified");
+                    continue;
                 }
-                Err(e) => {
-                    error!("Failed to load the new configuration: {:?}", e);
-                    warn!("Run Crusher with the previous config");
-                    fs::remove_file(&temp_path).unwrap_or_else(|e| {
-                        error!("Failed to remove the temporary file: {e}");
-                    });
+                let Ok(new_settings) = Settings::from_str(&config) else {
+                    error!("Failed to parse the configuration from the Manager server");
                     continue;
+                };
+                match (log_manager.change_log_dir)(
+                    settings.log_dir.as_deref(),
+                    new_settings.log_dir.as_deref(),
+                ) {
+                    Ok(Some(guard)) => log_manager.guard = guard,
+                    Err(e) => error!("Failed to update the log directory: {e}"),
```

Comment:
@danbi2990 
It seems that the following requirement might not be fully addressed. Could you kindly double-check this?

> For remote configurations: If log_dir is set but not writable, it should fall back to minimal mode and notify the server. In this minimal mode, if a log needs to be written, it should default to stdout.

Please let me know your thoughts. Thank you! 😊

Reply:
I think it might be a good idea to switch to stdout mode at this point, specifically when create_dir_all() or OpenOptions::new() fails. What do you think? 😊

Reply:
@danbi2990 
Firstly, I just realized that the logging policy I copy-pasted includes content that goes beyond the current scope of work. My apologies for any confusion.

> If log_dir is set but not writable, it should fall back to minimal mode and notify the server.
> Because if change_log_dir fails, the system will continue using the previous log_dir.

Ah, I understand now. It seems we need to align our perspectives on this!
I initially thought that "operating with the previous writer" was not included in the current logging policy.

**Please note that the following is entirely off-topic and just my personal opinion**:
I believe the current logging policy operates as follows, which is theoretically stateless:
```javascript
(default + arguments + remote configuration) -> logging policy behavior
```
On the other hand, the logging policy that @Jake mentioned seems to operate like this, which I think is theoretically stateful:
```javascript
(default + arguments + remote configuration, previous state) -> logging policy behavior
```
Personally, I've had some challenging experiences with behaviors determined by state machines, so I might have naturally tended to avoid them. 😇
However, I understand that referring to the "previous state" implies aiming for a **stable state**.

Reply:
@danbi2990 
I will close this review for now, as per your suggestion, so that the work can be fully addressed in the next issue.
I'm sorry for any inconvenience. 😅

Reply:
Thank you for providing the state machine perspective.
I will take it into account in the next issue.